### PR TITLE
[lora] Add load option to LoRA adapter API

### PR DESCRIPTION
**Pull request by xyang16**
## Description ##

Add an additional `load` option to the register adapter API and update adapter API.

The reason is for this change is to keep consistent with other model servers like vllm and lorax. vllm load_lora_adapter API don't load adapter weights, the adapter weights is only loaded when running inference using one particular adapter.

Discussed with Hosting team, making default to `true`.

Example:

```
curl -X POST "http://localhost:8080/models/model/adapters?name=eng_alpaca&load=false&src=/opt/ml/model/adapters/eng_alpaca"
```

### CODE REVIEW

File: engines/python/setup/djl_python/huggingface.py [hunk 1]
```diff
@@ -546,39 +575,45 @@ def update_adapter(inputs: Input):
...
-            if not pinned:
-                raise RuntimeError(
-                    f"Failed to pin LoRA adapter {adapter_alias}")
+        old_adapter_load = old_adapter.get_as_string("load").lower(
+        ) == "true" if old_adapter.contains_key("load") else True
+        if adapter_load != old_adapter_load:
+            if adapter_load:
+                _service.add_lora(adapter_name, adapter_alias, adapter_path)
+            else:
+                _service.remove_lora(adapter_name, adapter_alias)
```
Comment (siddvenk) [inline engines/python/setup/djl_python/huggingface.py:605]: if adapter_load is false, why are we removing the adapter? Should this just be a noop?

File: engines/python/setup/djl_python/rolling_batch/lmi_dist_rolling_batch.py [hunk 2]
```diff
@@ -283,5 +283,5 @@ def pin_lora(self, lora_name):
         # 2) An adapter is not evicted, call add_lora() is not necessary.
         # But since whether an adapter is evicted is not exposed outside of engine,
         # and add_lora() in this case will take negligible time, we will still call add_lora().
-        self.engine.add_lora(lora_request)
-        return self.engine.pin_lora(lora_request.lora_int_id)
+        return self.engine.add_lora(lora_request) and self.engine.pin_lora(
+            lora_request.lora_int_id)
```
Comment (siddvenk) [inline engines/python/setup/djl_python/rolling_batch/lmi_dist_rolling_batch.py:287]: do we need to check the result of add_lora before pinning?

Also, I would prefer if we kept these calls separate. It's more readable.

Sames questions for vlm_rolling_batch.py

File: engines/python/setup/djl_python/huggingface.py (hunk 1 above)
Comment (xyang16) [reply engines/python/setup/djl_python/huggingface.py:605]: This is to have an unload option.

File: engines/python/setup/djl_python/rolling_batch/lmi_dist_rolling_batch.py (hunk 2 above)
Comment (xyang16) [reply engines/python/setup/djl_python/rolling_batch/lmi_dist_rolling_batch.py:287]: 1) This is to make sure it's successfully loaded before pinning.
2) Made these calls separate.

File: engines/python/setup/djl_python/huggingface.py (hunk 1 above)
Comment (siddvenk) [reply engines/python/setup/djl_python/huggingface.py:605]: I see, so we support unloading only for unpinned adapters

Comment (xyang16) [reply engines/python/setup/djl_python/huggingface.py:605]: Yes

**Review by siddvenk — APPROVED** [review]

**Pull request merged**